test/tictactoe/Board_test.exs

```diff
@@ -0,0 +1,106 @@
+defmodule BoardTests do
+  use ExUnit.Case
+
+  describe "available spaces" do
+    test "returns empty list if there are no Empty spaces" do
+      board = [:NotEmpty, :NotEmpty]
+
+      assert Board.available_spaces(board) == []
+    end
+
+    test "returns list of indexes of all empty spaces" do
+      board = [:NotEmpty, :NotEmpty, :EMPTY, :EMPTY, :NotEmpty]
+
+      assert Board.available_spaces(board) == [2, 3]
+    end
+  end
+
+  describe "move result" do
+    test "player has won - returns :PlayerWon and resulting board" do
+      board = Fixtures.randomBoard()
+      has_player_won = true
+
+      assert Board.move_result(has_player_won, board) == {:PlayerWon, board}
+    end
+
+    test "player didn't win - returns :Playing and resulting board" do
+      board = Fixtures.randomBoard()
+      has_player_won = false
+
+      assert Board.move_result(has_player_won, board) == {:Playing, board}
+    end
+
+    test "player didn't win but there is a draw - returns :Playing and resulting board" do
+      board = Fixtures.randomBoard()
+      is_draw = true
+
+      assert Board.move_result(:Playing, is_draw, board) == {:Draw, board}
+    end
+
+    test "player didn't win and there is no draw - returns :Playing and resulting board" do
+      board = Fixtures.randomBoard()
+      is_draw = false
+
+      assert Board.move_result(:Playing, is_draw, board) == {:Playing, board}
+    end
+  end
+
+  describe "apply move" do
+    test "empty board gets updated and round continues" do
+      board = Fixtures.emptyBoard
+      move = Fixtures.randomSpaceIndex
+      player = Fixtures.randomPlayerSpace
+
+      expectedBoard = List.replace_at(board, move, player)
+
+      assert Board.apply_move(board, move, player) == {:Playing, expectedBoard}
```

Comment:
Why do we need this version of apply-move without the first argument?

Reply:
This one is used by `BotIntellect` it has no info about current round state.
https://github.com/SergiiVolodkoWorking/TicTacToeOnline/pull/13/files#diff-2482db705e17089fb9681c228cc22d90

Somehow, both those methods `Board.apply_move` and `BotIntellect.get_winning_move` are a bit alike in terms of having 2 versions of the method with and without the first argument. Arguments are different, but there is probably some coupling